x/valset/keeper/keeper.go

```diff
@@ -673,26 +712,19 @@ func (k Keeper) externalChainInfoStore(ctx sdk.Context, val sdk.ValAddress) sdk.
 	)
 }
 
-func (k Keeper) _externalChainInfoStore(ctx sdk.Context) sdk.KVStore {
-	return prefix.NewStore(
-		ctx.KVStore(k.storeKey),
-		[]byte("external-chain-info"),
-	)
+func (k Keeper) _externalChainInfoStore(ctx context.Context) storetypes.KVStore {
+	store := runtime.KVStoreAdapter(k.storeKey.OpenKVStore(ctx))
+	return prefix.NewStore(store, []byte("external-chain-info"))
 }
 
-func (k Keeper) snapshotStore(ctx sdk.Context) sdk.KVStore {
-	k.Logger(ctx).Debug("snapshot store", "store-key-name", k.storeKey.Name(), "store-key-string", k.storeKey.String())
-	return prefix.NewStore(ctx.KVStore(k.storeKey), []byte("snapshot"))
+func (k Keeper) snapshotStore(ctx context.Context) storetypes.KVStore {
+	store := runtime.KVStoreAdapter(k.storeKey.OpenKVStore(ctx))
+	// k.Logger(ctx).Debug("snapshot store", "store-key-name", k.memKey.Name(), "store-key-string", k.storeKey) //TODO
```

Comment:
TODO?

Reply:
We have removed the `memkey` storage because we are not using it anywhere ,so here in logger we have `memkey.name` and we need to remove it .That's why we have added TODO here.

Reply:
Resolve the TODO.

Reply:
Resolved